src/Components/Components/src/PersistentComponentState.cs

```diff
@@ -56,6 +56,11 @@ public PersistingComponentStateSubscription RegisterOnPersisting(Func<Task> call
     {
         ArgumentNullException.ThrowIfNull(callback);
 
+        if (PersistingState)
+        {
+            throw new InvalidOperationException("Registering a callback while persisting state is not allowed.");
```

Comment:
I don't understand. When `PersistingState` is `true`, then it means "persisting state is not allowed."? It's a bit confusing.

Reply:
I get it, it's rather "persisting state is already registered, another registration is not allowed". Right?

Reply:
Registering an additional callback is not allowed while we are persisting state, callbacks get registered ahead of time.